docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,188 @@
+        - [Module 5 Link Failures](#module-5-link-failures)
+        - [Module 6 Critical Process Crash](#module-6-critical-process-crash)
+        - [Module 7 Power down and hardware failure](#module-7-power-down-and-hardware-failure)
+        - [Module 8 Operations](#module-8-operations)
+    - [Test Utilities](#test-utilities)
+
+<!-- /TOC -->
+
+## Scope 
+This document proposes solutions for Smart Switch High-Availability test plans. The document will cover smart switch test scenarios.
+
+The goal of this test plan is to verify HA state machine behavior in normal operation scenarios and network failure scenarios with [t1-smartswitch-ha](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/vars/topo_t1-smartswitch-ha.yml) topology. Both control plane and data plane will need to be verified in the test cases. 
+
+## Topology
+
+### Physical Connection
+Tests will run on a sonic-mgmt testbed. The key physical components will be:
+1. Test server
+1. Fanout switches
+    * root fanout switch
+    * leaf fanout switch
+1. SONiC DUTs -- the two smartswitches.
+
+
+Key aspects of the physical connection:
+1. Every DUT port is connected to the leaf fanout switch.
+1. Every leaf fanout switch has unique VLAN tag for every DUT port.
+1. Root fanout switch connects leaf fanout switches and test servers using 802.1Q trunks. 
+
+For HA testbeds, the 2 smartswitch DUTs will be connected to the same leaf fanout, to eliminate factors that are irrelevant to the test scenarios. 
+
+Any test server can access any DUT port by sending a packet with the port VLAN tag. In test servers, a set of dockers can be created for running cEOS to simulate neighbors of the SONiC DUT. A PTF container (based on the PTF test framework) for injecting/sniffing packets will be created. The PTF docker, the cEOS dockers and VLAN interfaces in test server can be interconnected by open vSwitch bridges. 
+
+For HA testbeds, the 2 smartswitch DUTs will share the same set of cEOS dockers as the neighbors, and same PTF docker for packets injecting and sniffing. 
+
+![physical-connection](./Img/topo_smartswitch_ha.png)
+
+Please refer to [sonic-mgmt testbed overview](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.testbed.Overview.md) for more about testbed setup and diagrams, also [HA Topology](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md#4-network-physical-topology) for HA Network Physical Design. 
+
+### Setup configuration  
+
+The production scenario simulated with this testbed, is a VM-to-VM traffic scenario. Basically 2 VMs located in perhaps two different clusters in the data center, try to communicate with each other. Traffic passes through the HA set under test. Assuming dpu0 in SmartSwitch0 will be the Active node and dpu0 in SmartSwitch1 shall be set to Standby. Both DPUs will share same network configurations.
```

Comment:
for the scenario that we support in HA testing, it will be better to move it after the logical connection. and we can also indicate we will add more things in the future.

Reply:
Moved the test scenario after the diagrams.